clusterpools/apply.sh

```diff
@@ -510,14 +594,88 @@ if [[ "$CLUSTERPOOL_NAME" == "" ]]; then
     if [[ "$INPUT_CLUSTERPOOL_NAME" == "" ]]; then
         CLUSTERPOOL_NAME="$GENERATED_CLUSTERPOOL_NAME-$(echo $PLATFORM | tr '[:upper:]' '[:lower:]')-$(echo $CLUSTERIMAGESET_NAME | sed 's/\.//g')"
     else
+        CUSTOM_CLUSTERPOOL_NAME="true"
         CLUSTERPOOL_NAME="$INPUT_CLUSTERPOOL_NAME"
     fi
 fi
 printf "${GREEN}* Using Name: $CLUSTERPOOL_NAME${CLEAR}\n"
 
 
+#-----OPTIONALLY CONFIGURE A CUSTOM INSTALL CONFIG SECRET-----#
+if [[ "$YQ_INSTALLED" != "false" ]]; then
+    if [[ "$CUSTOM_CLUSTERPOOL_NAME" == "true" ]]; then
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-${RANDOM_IDENTIFIER}-install-config-template"
+    else
+        CLUSTERPOOL_INSTALL_CONFIG_SECRET_NAME="${CLUSTERPOOL_NAME}-install-config-template"
+    fi
```

Comment:
Wait. If I give a custom name, you're tagging a random identifier on it? That's not correct, is it?

Reply:
I'm appending a random identifier to the secret name in this case, not the clusterpool.  In this specific case, the user wants us to create an install-config-secret for them for a clusterpool that has a custom name.  If we create the secret without a random identifier and they try to re-create the same pool (after delete) without deleting the secret, or if they try to update the pool by rerunning, the process will fail because secrets can't be edited.  Adding the random identifier prevents secret collisions.

Reply:
Ah, that makes sense. Because if it's not custom, then it already has the identifier.
...but secrets can be updated, right? I feel like this goes against other flows where we overwrite ClusterClaims with reckless abandon.

Reply:
@dhaiducek well.... secrets can't _exactly_ be updated.  Its tricky to be honest.  Re-creating was much, much easier